src/archive.js

```diff
@@ -16,36 +23,45 @@ class Archive {
    * parent.
    * @param {Action} action Action to create an initial snapshot for
    */
-  create(action) {
+  create(action: Action): void {
     this.set(action, this.get(action.parent))
   }
 
   /**
    * Access a prior snapshot for a given action
    * @param {Action} action Action for requested snapshot
    */
-  get(action, fallback) {
+  get(action: Action, fallback: ?Payload): Payload {
     console.assert(action, 'Unable to get ' + action + ' action')
 
-    let value = this.pool[action.id]
+    if (this.has(action)) {
+      return this.pool[action.id]
+    }
 
-    return value === undefined ? fallback : value
+    return fallback == null ? {} : fallback
+  }
+
+  /**
+   * Does a snapshot exist for an action?
+   */
+  has(action: Action): boolean {
+    return typeof this.pool[action.id] !== 'undefined'
   }
 
   /**
    * Assign a new snapshot for an action
    * @param {Action} action Action for requested snapshot
    * @param {Object} snapshot
    */
-  set(action, snapshot) {
+  set(action: Action, snapshot: Payload) {
```

Comment:
You have `remove()` returning `void` but this unspecified. Does flow automatically assume `void` as a return value or should we add that here or take it off of `remove()`